docs/shrc_diag_util.md

```diff
@@ -22,104 +22,124 @@
+In the following example output, a populated cache is being accessed to look for compiled AOT code. Some AOT code is found, which can be shared, and some AOT code is stored for reuse.
 
-        1: 0x000000002237C6E0 STARTUP HINTS KEY: -Xoptionsfile=jre\bin\compressedrefs\options.default -Xlockword:mode=default -Xjcl:jclse29 -Dcom.ibm.oti.vm.bootstrap.library.path=jre\bin\compressedrefs;jre\bin -Djava.home=jre -Djava.ext.dirs=jre\lib\ext -Duser.dir=bin -Djava.class.path=. -Dsun.java.launcher=SUN_STANDARD Address: 0x000000002237C700 Size: 96
-        STARTUP HINTS DETAIL Flags: 1 DATA1: 1677721 DATA2: 5033165
+```
+java -Xshareclasses:name=myCache,verboseAOT HelloWorld
+[-Xshareclasses AOT verbose output enabled]
+Found AOT code for ROMMethod 0x00007F658005C180 in shared cache.
+Found AOT code for ROMMethod 0x00007F65800723EC in shared cache.
+Found AOT code for ROMMethod 0x00007F6580071D14 in shared cache.
+Stored AOT code for ROMMethod 0x00007F65801847B8 in shared cache.
+Stored AOT code for ROMMethod 0x00007F65800D38A4 in shared cache.
+Stored AOT code for ROMMethod 0x00007F65800723CC in shared cache.
+Found AOT code for ROMMethod 0x00007F65800D38A4 in shared cache.
+Stored AOT code for ROMMethod 0x00007F65800724C4 in shared cache.
+...
+```
+
+### `verboseHelper`
 
+To troubleshoot problems with custom class loaders that use the Java `SharedClassHelper` API, specify the [`-Xshareclasses:verboseHelper`](xshareclasses.md#verbosehelper) suboption. Information messages and error messages are generated in the output, which can help you diagnose problems with finding or storing classes in the shared cache.
 
-## `printStats`
+The following example output shows only information messages:
+
+```
+java -Xshareclasses:name=myCache,verboseHelper HelloWorld
+[-Xshareclasses Helper API verbose output enabled]
+Info for SharedClassURLClasspathHelper id 1: Verbose output enabled for SharedClassURLClasspathHelper id 1
+Info for SharedClassURLClasspathHelper id 1: Created SharedClassURLClasspathHelper with id 1
+Info for SharedClassURLClasspathHelper id 2: Verbose output enabled for SharedClassURLClasspathHelper id 2
+Info for SharedClassURLClasspathHelper id 2: Created SharedClassURLClasspathHelper with id 2
+Info for SharedClassURLClasspathHelper id 1: There are no confirmed elements in the classpath. Returning null.
+Info for SharedClassURLClasspathHelper id 2: There are no confirmed elements in the classpath. Returning null.
+Info for SharedClassURLClasspathHelper id 2: setClasspath() updated classpath. No invalid URLs found
+Info for SharedClassURLClasspathHelper id 2: Number of confirmed entries is now 1
+Hello World
+```
+
+## Diagnostic cache utilities
+
+These utilities display information about the contents of a shared classes cache. Run the utilities by specifying them as suboptions of `-Xshareclasses`. The utilities run on the default cache unless you specify a cache by adding the `name=<cache_name>` suboption.
```

Comment:
We may also want to mention `cacheDir=<dir>` if user is not using the default cache directory.